src/main/java/com/example/userrestapi/controller/UserController.java

```diff
@@ -23,20 +27,37 @@
 
         List<User> users = new ArrayList<User>();
 
-        if (name != null) {
+        //I get it this is probably not the most optimal solution,
+        //as with more parameters the amount of conditions is going to raise drastically,
+        //but it's the only one I was able to come up wth right now
+
+        //FIXME find a better solution to this workaround
+        if ((name != null) && (surname != null) && (email != null)) {
+            users = repository.findByName(name).stream().filter(user -> user.getSurname().equals(surname)).collect(Collectors.toList()).
```

Comment:
you can use AND in the filter method
```suggestion
            users = repository.findByName(name).stream().filter(user -> user.getSurname().equals(surname) && user.getEmail().equals(email)).collect(Collectors.toList()).
```